src/cmds/qsub_sup.c

```diff
@@ -258,8 +261,21 @@ get_script(FILE *file, char *script, char *prefix)
 		return (4);
 	}
 
-	while ((in = pbs_fgets_extend(&s_in, &s_len, file)) != NULL) {
+	while ((in = pbs_fgets(&s_in, &s_len, file)) != NULL) {
 		if (!exec && ((sopt = pbs_ispbsdir(s_in, prefix)) != NULL)) {
+			/* Check if this is a directive line that should be extended */
+			extend_len = pbs_extendable_line(in);
+			if (extend_len >= 0) {
+				in[extend_len] = '\0'; /* remove the backslash (\) */
+				extend = pbs_fgets_extend(&extend_in, &extend_len, file);
```

Comment:
extend_len is the amount of space that is currently in extend_in.  At the start of the loop, extend_in == NULL, this will cause a crash.  If you don't preallocate any memory for extend_in, extend_len == 0.

Preallocating memory for extend_in is unnecessary.  pbs_fgets_extend() will do it for you.

Reply:
New push corrects to use `extend_loc` to save location from `pbs_extendable_line`
Also some corrections to memory management, ensuring that `pbs_fgets_extend` return is freed, etc.